frame/composable-traits/src/dex.rs

```diff
@@ -111,3 +112,28 @@ pub struct ConstantProductPoolInfo<AccountId, AssetId> {
 	/// Amount of the fee pool charges for the exchange
 	pub fee: Permill,
 }
+
+#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum DexRouteNode<PoolId> {
+	Curve(PoolId),
+	Uniswap(PoolId),
+}
+
+/// Describes route for DEX.
+/// `Direct` gives vector of pool_id to use as router.
+#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum DexRoute<PoolId, MaxHops: Get<u32>> {
+	Direct(BoundedVec<DexRouteNode<PoolId>, MaxHops>),
+}
+
+pub trait DexRouter<AssetId, PoolId> {
+	/// If route is `None` then delete existing entry for `asset_pair`
+	/// If route is `Some` and no entry exist for `asset_pair` then add new entry
+	/// else update existing entry.
+	fn update_route(
```

Comment:
May be use CurrencyPair?

Reply:
`CurrencyPair` has `base` and `quote` but usually these should be able to use symmetrically.